plugins/source_scheduler_dfs.go

```diff
@@ -71,7 +73,21 @@ func (p *SourcePlugin) syncDfs(ctx context.Context, spec specs.Source, client sc
 			}()
 		}
 	}
+
+	// We start a goroutine that logs the metrics periodically.
+	// It needs its own waitgroup
+	var logWg sync.WaitGroup
+	logWg.Add(1)
+
+	logCtx, logCancel := context.WithCancel(ctx)
+	go p.periodicMetricLogger(logCtx, &logWg)
```

Comment:
you want to start this before you start spawning all the clients as it might be blocked if there are many tables/clients.
After you run `go p.periodicMetricLogger` you should just call `defer logCancel()`